.github/workflows/unit-tests.yaml

```diff
@@ -0,0 +1,50 @@
+name: Unit Tests
+
+on:
+  pull_request:
+    branches: [ main, release-* ]
+
+jobs:
+  unit-tests:
+    name: Unit Tests
+    runs-on: ubuntu-latest
+    steps:
+    - name: Free some disk space on runner
+      run: |
+        echo "free space before cleanup:"
+        df -h
+        sudo rm -rf /usr/share/dotnet /opt/ghc /usr/local/share/boost /usr/lib/jvm /usr/lib/firefox /opt/microsoft/powershell /opt/hostedtoolcache
+        echo "free space after cleanup:"
+        df -h
+
+    - name: Set up Go 1.x
+      uses: actions/setup-go@v3
+      with:
+        go-version: 1.19
+      id: go
+
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@v1
+
+    - name: go cache
+      uses: actions/cache@v3
+      with:
+        path: |
+          ~/.cache/go-build
+          ~/go/pkg/mod
+        key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+        restore-keys: |
+          ${{ runner.os }}-go-
+
+    - name: Get dependencies
+      run: go mod download
+
+    - name: Run make test
+      run: |
+        make test COMPONENT=featuregates/controller
+        make test COMPONENT=capabilities/controller
+
+    - name: Upload coverage reports to Codecov with GitHub Action
+      uses: codecov/codecov-action@v3
+      with:
+        files: ./build/capabilities/controller/coverage/cover.out,./build/featuregates/controller/coverage/cover.out
```

Comment:
you should also pick up coverage files from client modules([featuregates client](https://github.com/vmware-tanzu/tanzu-framework/tree/main/featuregates/client), [capabilities client](https://github.com/vmware-tanzu/tanzu-framework/tree/main/capabilities/client))

Reply:
It's done in the latest commit.